containers/Connection.ts

```diff
@@ -1,54 +1,64 @@
 import { createContainer } from "unstated-next";
 import { useState, useEffect } from "react";
 import { ethers } from "ethers";
+import { API as OnboardApi, Wallet } from "bnc-onboard/dist/src/interfaces";
+import Onboard from "bnc-onboard";
 import { Observable } from "rxjs";
 import { debounceTime } from "rxjs/operators";
 
+import { config } from "./Config";
+
 type Provider = ethers.providers.Provider;
 type Block = ethers.providers.Block;
 type Network = ethers.providers.Network;
-type ExternalProvider = ethers.providers.ExternalProvider;
 type Signer = ethers.Signer;
 
 function useConnection() {
   const [provider, setProvider] = useState<Provider | null>(null);
+  const [onboard, setOnboard] = useState<OnboardApi | null>(null);
   const [signer, setSigner] = useState<Signer | null>(null);
   const [network, setNetwork] = useState<Network | null>(null);
   const [address, setAddress] = useState<string | null>(null);
+  const [selectedWallet, setSelectedWallet] = useState<string | null>(null);
   const [error, setError] = useState<Error | null>(null);
   const [block$, setBlock$] = useState<Observable<Block> | null>(null);
 
   const attemptConnection = async () => {
-    if (window.ethereum === undefined) {
-      throw Error("MetaMask not found, please visit https://metamask.io/");
-    }
-
-    // // for testing
-    // const provider = new ethers.providers.JsonRpcProvider()
-    // let signer = new ethers.Wallet(process.env.PRIV_KEY)
-    // signer = signer.connect(provider)
-
-    // get provider and signer
-    const provider = new ethers.providers.Web3Provider(
-      window.ethereum as ExternalProvider
-    );
-    const signer = provider.getSigner();
-    const network = await provider.getNetwork();
+    const onboard = Onboard({
```

Comment:
Wait, why do you declare `onboard` here but also in the `useState()` on line 18?

Reply:
the reason is that I need to full set up the onboard instance with `await onboardInstance.walletSelect();` before setting into the container hook. I have re-named this variable to make it a bit clear that this is the `onboardInstance`..